src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1290,108 +1291,100 @@ private bool ShouldContextualKeywordBeTreatedAsModifier(bool parsingStatementNot
             // Some of our helpers start at the current token, so we'll have to advance for their
             // sake and then backtrack when we're done.  Don't leave this block without releasing
             // the reset point.
-            ResetPoint resetPoint = GetResetPoint();
+            using var _ = GetDisposableResetPoint(resetOnDispose: true);
```

Comment:
`resetOnDispose: true` is used for the pattern where the finally is:

```c#
            finally
            {
                this.Reset(ref resetPoint);
                this.Release(ref resetPoint);
            }
```

`resetOnDispose: false` is used for the pattern where the finally is:

```c#
            finally
            {
                this.Release(ref resetPoint);
            }
```

Reply:
most of the time we can use `_` which indicates that we unilaterally reset/release when we drop out of scope.  When this has a name it's because the body code has custom logic about when to reset.

Reply:
This doc could be added as a comment on `GetDisposableResetPoint`